nucypher/utilities/nodedeploy.py

```diff
@@ -0,0 +1,426 @@
+import os
+import json
+import maya
+import time
+import hashlib
+from base64 import b64encode
+
+from jinja2 import Template, TemplateError
+
+from botocore import exceptions as botoexceptions
+
+from nucypher.config.constants import DEFAULT_CONFIG_ROOT
+from nucypher.blockchain.eth.clients import PUBLIC_CHAINS
+from nucypher.blockchain.eth.networks import NetworksInventory
+from nucypher.blockchain.eth.registry import BaseContractRegistry
+
+NODE_CONFIG_STORAGE_KEY = 'worker-configs'
+URSULA_PORT = 9151
+PROMETHEUS_PORTS = [9101, 9090]
+
+def preflight_check(*attrs):
+    necessary_attrs = attrs
+    def wrapper(fn):
+        def wrapped(self, *args, **kwargs):
+            if all([getattr(self, a, False) for a in necessary_attrs]):
+                return fn(self, *args, **kwargs)
+        return wrapped
+    return wrapper
+
+
+class CloudNodeConfigurator:
+    """
+    configures an existing running node
+    """
+
+    clef_rules_template = """
```

Comment:
Personally, I'd prefer this as a formal template, perhaps jinja2 in it's own file.